### PR TITLE
Animated WebP Support + Memory Optimisations

### PR DESCRIPTION
**Pull request by MichaelOultram**
* Create ImgBuf enum to hold File/Stdin reader
* When animated, only decode the first frame for the make_img_request
* Change compress_frames to take frames iter instead of a GifDecoder

### CODE REVIEW

**Comment by Horus645** [comment]
Oooh! This should close #111.

Thanks!

Overall, this LGTM, let me just test it for little bit before merging.

**Comment by Horus645** [comment]
Yep, this is really great, you even cleaned up some of the code.

Thanks again!

**Comment by beh-10257** [comment]
is decoding the first frame only still necessary ??
from what I can see here
https://github.com/image-rs/image/issues/1550
its already added so why not enable decoding the whole thing

**Comment by MichaelOultram** [comment]
> When animated, only decode the first frame for the make_img_request

@beh-10257: This text might be slightly misleading; it was an implementation detail referring to a memory optimisation not a limitation.

There are two independent IPC requests for GIF/WebP. `make_img_request` only needs the first frame (previously it was decoding every frame) and `make_animation_request` needs every frame. As these requests are independent, they were unnecessarily decoding every frame twice which consumed a lot of memory.

TLDR: Animated WebP images should show the animation correctly at the time of this commit. I've not yet tried the latest release to check if it's still working.

**Comment by beh-10257** [comment]
@MichaelOultram can you try this in your system
[output.txt](https://github.com/LGFae/swww/files/14876227/output.txt)
rename it to .webp I am just getting around github limitations here
for me its not working on latest release 0.9.1

**Comment by LGFae** [comment]
We need to decode just the first frame to play out the transition from the previous image to the current one. The animation then starts after that. Changing that would be pretty involved.

This is also not the place to file for bug reports. If you are having problems with `webp` images, you should open an issue.

**Comment by beh-10257** [comment]
I made an issue https://github.com/LGFae/swww/issues/260
sorry for the annoyance once again
and FWIW I really though the animation not showing up was by design that why I came here sorry once again
